commit 3b73dab1a9e676e28334df10fac7c054418228a8
Author: Michael Schroeder <mls@suse.de>
Date:   Fri Mar 17 10:49:14 2017 +0100

    [backend] fix kiwitree symlink check

    Bad code copied from the build package. Sigh.

Origin: upstream, https://github.com/openSUSE/open-build-service/commit/3b73dab1a9e676e28334df10fac7c054418228a8
--- a/src/backend/bs_repserver
+++ b/src/backend/bs_repserver
@@ -1743,7 +1743,7 @@ sub receivekiwitree {
     } elsif ($type eq 'l') {
       $extra =~ s/%([a-fA-F0-9]{2})/chr(hex($1))/ge;
       die("bad symlink\n") if "/$extra/" =~ /\/\.?\//;
-      if ("/$extra/" =~ /^(\/\.\.)+\/(.*?)$/s) {
+      if ("/$extra/" =~ /^((?:\/\.\.)+)\/(.*?)$/s) {
         my ($head, $tail) = ($1, $2);
 	die("bad upref in symlink\n") if "/$tail/" =~ /\/\.\.\//;
 	die("bad upref in symlink\n") if ($head =~ y!/!!) > ($file =~ y!/!!);
